pkg/apis/v1beta1/nodeclaim.go

```diff
@@ -66,6 +66,8 @@ type NodeClaimSpec struct {
 	//
 	// The feature can also be used to allow maximum time limits for long-running jobs which can delay node termination with preStop hooks.
 	// If left undefined, the controller will wait indefinitely for pods to be drained.
+	// +kubebuilder:validation:Pattern=`^([0-9]+(\\.[0-9]+)?(ms|s|m|h|d|w|y))+$`
```

Comment:
I discussed with @wmgroot, but we intentionally only keep it as s/m/h, due to inconsistencies in how days/weeks/years are calculated.  @cnmcavoy can you add that back in as it was? 

https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1/nodepool.go#L84

Reply:
I'm testing out the branch with day+ removed.

Reply:
Should be all good now.